src/digest.rs

```diff
@@ -114,7 +114,8 @@ impl BlockContext {
 
         Digest {
             algorithm: self.algorithm,
-            value: (self.algorithm.format_output)(self.state),
+            // SAFETY: TODO
```

Comment:
PR #1731 will (I hope) address this as well, by making the code 100% safe as long as we are happy with the safety guarantees for the code that writes to the `self.state` union.

Which functions within `digest` need to be marked `unsafe` and which can be safe with `unsafe` blocks is something that could use discussion as it seems we may disagree, which means I might be misunderstanding something.